assignments/week_2/project_2/readme.md

```diff
@@ -0,0 +1,135 @@
+sensor inputs. Each light state will last for a certain duration, and the controller will only need to check the sensors
+to decide which direction to turn green. There are no turns allowed at this intersection, so we don't need to worry
+about turn signals.
+
+## Program Behavior
+
+Here is a summary of our desired light behavior:
+
+* All lights should be red for one second between cycles.
+* All yellow lights should stay lit for 2 seconds
+* All green lights should stay lit for 5 seconds
+* Green lights should only be turned on if their direction was just red
+* Yellow lights should only be turned on if their direction was just green
+* Red lights should only be turned on if their direction was just yellow
+
+Each second of simulated time is indicated by a new line of input sent to the program's standard input. Each line of
+input contains two binary digits. The first digit is the reading from the north sensor, and the second digit is the
+reading from the south sensor.
+
+Your program should respond to each line of input with a line of output describing which lights your controller would
+like to turn on. This output should be six binary digits. These digits should encode the light outputs as follows
+
+1. Red lights for north-south road
+2. Yellow lights for north-south road
+3. Green lights for north-south road
+4. Red lights for east-west road
+5. Yellow lights for east-west road
+6. Green lights for east-west road
+
+In this encoding scheme, red in all directions is "100100". Green for east-west is "100001", and green for north-south
+is "001100".
+
+A full cycle that allows the east-west road to travel would look like this:
+
+100100 // Start cycle with all red lights
+100001 // Green lights for east-west
+100001
+100001
+100001
+100001  // Green lights have now been on for 5 seconds
+100010  // Yellow lights for east-west
+100010  // Yellow lights have now been on for 2 seconds
+100100  // Back to red to complete the cycle
+
+When the controller is ready to turn on green lights to start a cycle, it must check the car sensors. If either sensor
+is reading high (1), the controller should turn on the green lights for the north-south road. If both sensors are low
+(0), the controller should turn on the green lights for the east-west road.
+
+## State Machine Design
+
+Before you start writing any code, design the state machine that your controller program will be implementing. Think
+about what states your controller needs to have. What transitions need to exist between these states? How can you keep
+track of time in this state machine?
+
+We've included an example solution state machine here, but make sure you take time to try to work through this part
+yourself. This is here to help you along if you really, truly get stuck.
+
+<details>
+<summary>Solution</summary>
+
+![Example state machine](state_machine.png)
+
+</details>
```

Comment:
Maybe have two levels of hint, where the first one tells you what the states should be and the second one is the full state machine?